proto-lens-tests/tests/partial_test.hs

```diff
@@ -0,0 +1,98 @@
+-- Copyright 2016 Google Inc. All Rights Reserved.
+--
+-- Use of this source code is governed by a BSD-style
+-- license that can be found in the LICENSE file or at
+-- https://developers.google.com/open-source/licenses/bsd
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TypeApplications #-}
+module Main where
+
+import qualified Data.Map as M
+import Data.ProtoLens
+import Data.ProtoLens.Arbitrary
+import Data.ProtoLens.Compare (partial)
+import Lens.Family2 ((&), (.~))
+import Test.Framework (testGroup)
+import Test.Framework.Providers.HUnit (testCase)
+import Test.HUnit.Base (assertBool)
+
+import Data.ProtoLens.TestUtil
+import Proto.Partial
+import Proto.Partial'Fields
+
+defFoo :: Foo
+defFoo = def
+
+main :: IO ()
+main = testMain
+    [ testAlwaysMatchDefault
+    , testAlwaysMatchSame
+    , testMatchRequired
+    , testMatchOptional
+    , testMatchRepeated
+    ]
+
+makeTestCase :: Message a => String -> a -> a -> Test
+makeTestCase testName against match = testCase testName $ do
+    assertBool "matched"         $ partial against match
+    assertBool "not const" . not $ partial match against
```

Comment:
What does "not const" mean?

Reply:
it means `partial` isn't implemented as `\_ _ -> True`  :)